composer/metrics/nlp.py

```diff
@@ -287,6 +289,75 @@ def update(self, batch: dict, output_logits: torch.Tensor, labels: torch.Tensor)
+    match one of the possible answer aliases (referred to as the 'continuation').
+
+    For example, the model may be provided the context below and evaluated on its ability to correctly predict the continuation.
+
+    Context: `Question: Who was president of the United States in 2012?\nAnswer: Barack Obama\nQuestion: Is water wet?\nAnswer: `
+    Continuation: [`yes`, `no`]
+
+    Both predictions and answers will be normalized before comparison.
+
+    Adds metric state variables:
+        correct (float): The number of instances where the prediction was a prefix for any of the answer aliases.
+        total (float): The number of total instances that were predicted.
+
+    Args:
+        dist_sync_on_step (bool, optional): Synchronize metric state across processes at
+            each forward() before returning the value at the step. Default: ``False``.
+    """
+
+    # Make torchmetrics call update only once
+    full_state_update = False
+
+    def __init__(self, dist_sync_on_step: bool = False):
+        # state from multiple processes
+        super().__init__(dist_sync_on_step=dist_sync_on_step)
+        self.add_state('correct', default=torch.tensor(0.), dist_reduce_fx='sum')
+        self.add_state('total', default=torch.tensor(0.), dist_reduce_fx='sum')
+
+    def normalize_answer(self, answer: str):
+        """Lower text and remove punctuation, articles and extra whitespace.
+
+        Copied from https://github.com/mandarjoshi90/triviaqa/blob/master/evaluation/triviaqa_evaluation.py
+        """
+
+        def remove_articles(text: str) -> str:
+            return re.sub(r'\b(a|an|the)\b', ' ', text)
+
+        def white_space_fix(text: str) -> str:
+            return ' '.join(text.split())
+
+        def handle_punc(text: str) -> str:
+            exclude = set(string.punctuation + ''.join([u'‘', u'’', u'´', u'`']))
+            return ''.join(ch if ch not in exclude else ' ' for ch in text)
+
+        def lower(text: str) -> str:
+            return text.lower()
+
+        def replace_underscore(text: str) -> str:
+            return text.replace('_', ' ')
+
+        return white_space_fix(remove_articles(handle_punc(lower(replace_underscore(answer))))).strip()
+
+    def update(self, outputs: List[str], labels: List[List[str]]):
+        for sample_output, sample_labels in zip(outputs, labels):
+            cleaned_sample_output = self.normalize_answer(sample_output)
+            cleaned_sample_labels = set(self.normalize_answer(label) for label in sample_labels)
+            if any(cleaned_sample_output.startswith(label) for label in cleaned_sample_labels):
+                self.correct += torch.tensor(1.0)
+            self.total += torch.tensor(1.0)
+
+    def compute(self):
+        assert isinstance(self.correct, Tensor)
+        assert isinstance(self.total, Tensor)
+        return self.correct / self.total
```

Comment:
Can we add + epsilon (and make epsilon an argument) just to be safe? iirc other metrics implementations do this? 

I'm also fine erroring 🤷

Reply:
This is what you get if you call compute before update (all of our metrics have this property)
```
In [2]: from composer.metrics import MaskedAccuracy

In [3]: acc = MaskedAccuracy()

In [4]: acc.compute()
/workdisk/danielking/miniconda3/envs/examples-llm/lib/python3.10/site-packages/torchmetrics/utilities/prints.py:36: UserWarning: The ``compute`` method of metric MaskedAccuracy was called before the ``update`` method which may lead to errors, as metric states have not yet been updated.
  warnings.warn(*args, **kwargs)
Out[4]: tensor(nan)
```

Reply:
So I will leave it